crates/sui-core/src/transaction_outputs.rs

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::collections::{HashMap, HashSet};
+use std::sync::Arc;
+use sui_types::base_types::ObjectRef;
+use sui_types::effects::{TransactionEffects, TransactionEffectsAPI, TransactionEvents};
+use sui_types::inner_temporary_store::{InnerTemporaryStore, WrittenObjects};
+use sui_types::storage::{MarkerValue, ObjectKey};
+use sui_types::transaction::{TransactionDataAPI, VerifiedTransaction};
+
+/// TransactionOutputs
+pub struct TransactionOutputs {
+    pub transaction: Arc<VerifiedTransaction>,
+    pub effects: TransactionEffects,
+    pub events: TransactionEvents,
+
+    pub markers: Vec<(ObjectKey, MarkerValue)>,
+    pub wrapped: Vec<ObjectKey>,
+    pub deleted: Vec<ObjectKey>,
+    pub locks_to_delete: Vec<ObjectRef>,
+    pub new_locks_to_init: Vec<ObjectRef>,
+    pub written: WrittenObjects,
+}
+
+impl TransactionOutputs {
```

Comment:
The refactoring of TransactionOutputs is a bit distracting in this PR.
Ideally (for future reference) you could make it a separate PR because it contains very sensitive logic, which is different from most of the other mechanical refactoring.